code/ship/ship.cpp

```diff
@@ -7393,6 +7412,10 @@ void ship_render_cockpit(object *objp)
 	gr_end_proj_matrix();
 
 	hud_save_restore_camera_data(0);
+
+	//Restore the Shadow_override
+	if (Shadow_quality != ShadowQuality::Disabled)
+		Shadow_override = true;
```

Comment:
This might have external effects. If you want to restore the previous state then you need to store that in a local variable before updating it in the block above.